server/src/domain/asset/response-dto/asset-response.dto.ts

```diff
@@ -71,15 +74,23 @@ function _map(entity: AssetEntity, withExif: boolean): AssetResponseDto {
     isExternal: entity.isExternal,
     isOffline: entity.isOffline,
     isReadOnly: entity.isReadOnly,
+    hasMetadata: true,
   };
 }
 
-export function mapAsset(entity: AssetEntity): AssetResponseDto {
```

Comment:
These used to be separate and it was a pain because every new property or change had to be made in two places and sometimes there were differences (on accident). Can you instead keep it as a single method with a boolean flag. The implementation can change to something like this though:

```typescript
const sanitized = { 
   ...
 };

const unsanitized = { 
  ...sanitized
  ...
 };

return withMetadata? unsanitized : sanitized;
```

Reply:
Done!

Reply:
Thank you!